filter/bytes_filter.go

```diff
@@ -0,0 +1,92 @@
+package filter
+
+import (
+	"bytes"
+
+	"github.com/xichen2020/eventdb/values/iterator"
+	xbytes "github.com/xichen2020/eventdb/x/bytes"
+)
+
+// BytesFilter matches against bytes values.
+type BytesFilter interface {
+	// Match returns true if the given value is considered a match.
+	Match(v iterator.Bytes) bool
+}
+
+type bytesToBytesFilterFn func([]byte) bytesFilterFn
+
+type bytesFilterFn func(v []byte) bool
+
+func (fn bytesFilterFn) Match(v iterator.Bytes) bool { return fn(v.Data) }
+
+func equalsBytesBytes(rhs []byte) bytesFilterFn {
```

Comment:
nit: rename to `equalBytesFn`

Reply:
This is consistent w/ the type specific naming of all filter fns.

```
boolToBoolFilterOps = map[Op]boolToBoolFilterFn{
        Equals:    equalsBoolBool,
        NotEquals: notEqualsBoolBool,
}
intToIntFilterOps = map[Op]intToIntFilterFn{
        Equals:             equalsIntInt,
        NotEquals:          notEqualsIntInt,
        LargerThan:         largerThanIntInt,
        LargerThanOrEqual:  largerThanOrEqualIntInt,
        SmallerThan:        smallerThanIntInt,
        SmallerThanOrEqual: smallerThanOrEqualIntInt,
}
doubleToIntFilterOps = map[Op]doubleToIntFilterFn{
        Equals:             equalsIntDouble,
        NotEquals:          notEqualsIntDouble,
        LargerThan:         largerThanIntDouble,
        LargerThanOrEqual:  largerThanOrEqualIntDouble,
        SmallerThan:        smallerThanIntDouble,
        SmallerThanOrEqual: smallerThanOrEqualIntDouble,
}
doubleToDoubleFilterOps = map[Op]doubleToDoubleFilterFn{
        Equals:             equalsDoubleDouble,
        NotEquals:          notEqualsDoubleDouble,
        LargerThan:         largerThanDoubleDouble,
        LargerThanOrEqual:  largerThanOrEqualDoubleDouble,
        SmallerThan:        smallerThanDoubleDouble,
        SmallerThanOrEqual: smallerThanOrEqualDoubleDouble,
}
intToDoubleFilterOps = map[Op]intToDoubleFilterFn{
        Equals:             equalsDoubleInt,
        NotEquals:          notEqualsDoubleInt,
        LargerThan:         largerThanDoubleInt,
        LargerThanOrEqual:  largerThanOrEqualDoubleInt,
        SmallerThan:        smallerThanDoubleInt,
        SmallerThanOrEqual: smallerThanOrEqualDoubleInt,
}
bytesToBytesFilterOps = map[Op]bytesToBytesFilterFn{
        Equals:             equalsBytesBytes,
        NotEquals:          notEqualsBytesBytes,
        LargerThan:         largerThanBytesBytes,
        LargerThanOrEqual:  largerThanOrEqualBytesBytes,
        SmallerThan:        smallerThanBytesBytes,
        SmallerThanOrEqual: smallerThanOrEqualBytesBytes,
        StartsWith:         startsWithBytesBytes,
        DoesNotStartWith:   doesNotStartWithBytesBytes,
        EndsWith:           endsWithBytesBytes,
        DoesNotEndWith:     doesNotEndWithBytesBytes,
        Contains:           containsBytesBytes,
        DoesNotContain:     doesNotContainBytesBytes,
}
timeToTimeFilterOps = map[Op]timeToTimeFilterFn{
        Equals:             equalsTimeTime,
        NotEquals:          notEqualsTimeTime,
        LargerThan:         largerThanTimeTime,
        LargerThanOrEqual:  largerThanOrEqualTimeTime,
        SmallerThan:        smallerThanTimeTime,
        SmallerThanOrEqual: smallerThanOrEqualTimeTime,
}
```